File: .NET PROJECT/Models/City.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace MedLab.Models
{
    public class City
    {
        [Key]
        public int CityId { get; set; }

        [Required(ErrorMessage = "City Name is required")]
        public string CityName { get; set; } = string.Empty;

        [Required(ErrorMessage = "State Id is required")]
        [ForeignKey("State")]
        public int StateId { get; set; }

        public State? State { get; set; }

        public ICollection<User> Users { get; set; } = new HashSet<User>();
    }
}

File: Springboot project/SPRINGBOOT PROJECT/MedLab/MedLab/src/main/java/com/medlab/models/City.java
package com.medlab.models;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import java.time.LocalDateTime;

@Entity
@Table(name = "cities")
public class City {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer cityId;

    @NotNull
    @Size(max = 100)
    private String cityName;


    @ManyToOne
    @JoinColumn(name = "state_id")
    private State state; // Navigation property

    
    

   
    // Getters and setters
    public Integer getCityId() {
        return cityId;
    }

    public void setCityId(Integer cityId) {
        this.cityId = cityId;
    }

    public String getCityName() {
        return cityName;
    }

    public void setCityName(String cityName) {
        this.cityName = cityName;
    }

  

 

    public State getState() {
        return state;
    }

    public void setState(State state) {
        this.state = state;
    }

   
   
}

